src/main/java/com/epam/training/olha_haichenkova/task_1/page/PastebinHomePage.java

```diff
@@ -50,23 +55,31 @@ public PastebinHomePage inputNewPasteTitle(String title){
     }
 
     public PastebinHomePage selectExpirationTime(String optionName) {
-        new WebDriverWait(driver, Duration.ofSeconds(WAIT_TIMEOUT_SECONDS))
-                .until(ExpectedConditions.elementToBeClickable(expirationTimeDropdown)).click();
+        wait.until(ExpectedConditions.elementToBeClickable(expirationTimeDropdown)).click();
         WebElement option = driver.findElement(By.xpath(createPathToExpirationTimeOption(optionName)));
         option.click();
         return this;
     }
 
     private static String createPathToExpirationTimeOption(String optionName){
-        String pathToOptionStart = "//ul[@id='select2-postform-expiration-results']/li[text()='";
-        String pathToOptionEnd = "']";
-        return pathToOptionStart + optionName + pathToOptionEnd;
+        return String.format(pathToOption, optionName);
     }
 
     public PastebinCreatedPastePage createNewPaste(){
-        new WebDriverWait(driver, Duration.ofSeconds(WAIT_TIMEOUT_SECONDS))
-                .until(ExpectedConditions.elementToBeClickable(createNewPasteButton)).click();
+        closeAdvertisementIfAppear();
+        wait.until(ExpectedConditions.elementToBeClickable(createNewPasteButton)).click();
         return new PastebinCreatedPastePage(driver);
     }
 
+    private void closeAdvertisementIfAppear() {
+        List<WebElement> advPopUp = driver.findElements(By.id(advertisementPopUpId));
+        if (!advPopUp.isEmpty()) {
+            closeAdvertisementPopUp();
+        }
```

Comment:
Are there really several elements with the same 'id' on the page? I believe that this 'if' block and list of elements not needed here

Reply:
I use a List to avoid NoSuchElementException since advertisements appear randomly

Reply:
In such case it's better just to use selenium check, if element displayed. We can create a method, that will perform the check for our popup, isDisplayed. And this method will be suitable for our 'if' statement